### ISSUE TITLE
Make node startup robust

### ISSUE THREAD

**Issue by Mirko-von-Leipzig** [0xPolygonMiden/miden-node#486]
The current `node` startup procedure is effectively

```rust
spawn(store);
sleep(1s);
spawn(block-producer);
sleep(1s);
spawn(rpc);
```

where `block-producer` depends on `store`, and `rpc` depends on `store`/both (?). This is hacky and should be replaced with a more robust startup system (see #483).

**Comment by bobbinth** [comment]
This is somewhat related to #73 - but also probably is a separate issue.

Im thinking that for now we can go with the option 2 proposed in https://github.com/0xPolygonMiden/miden-node/issues/483#issuecomment-2333825113 (i.e., components signal to the downstream components when they are ready).

We can also create a separate issue to handle start-up of the node as independent components - but for now I think we can say that it is the responsibility of the user to start them in sequence such that a downstream component is started only after an upstream component finished loading.

**Comment by bobbinth** [comment]
Let's fix this directly against `main` branch. This way, we can release `miden-node` v0.5.1 which would be the basis for the Alpha 4 testnet.
